test/project-commands.ts

```diff
@@ -0,0 +1,114 @@
+/// <reference path=".d.ts" />
+"use strict";
+
+import * as yok from "../lib/common/yok";
+import * as stubs from "./stubs";
+import { CreateProjectCommand } from "../lib/commands/create-project";
+import * as constants from "../lib/constants";
+import {assert} from "chai";
+
+let selectedTemplateName: string;
+let isProjectCreated: boolean;
+let dummyArgsString = "dummyArgsString";
+
+class ProjectServiceMock implements IProjectService {
+	createProject(projectName: string, selectedTemplate?: string): IFuture<void> {
+		return (() => {
+			selectedTemplateName = selectedTemplate;
+			isProjectCreated = true;
+		}).future<void>()();
+	}
+}
+
+class ProjectNameValidatorMock implements IProjectNameValidator {
+	public validate(name: string): boolean {
+		return true;
+	}
+}
+
+function createTestInjector() {
+	let testInjector = new yok.Yok();
+
+	testInjector.register("injector", testInjector);
+	testInjector.register("staticConfig", {});
+	testInjector.register("projectService", ProjectServiceMock);
+	testInjector.register("errors", stubs.ErrorsStub);
+	testInjector.register("logger", stubs.LoggerStub);
+	testInjector.register("projectNameValidator", ProjectNameValidatorMock);
+	testInjector.register("options", {
+		ng: false,
+		template: undefined
+	});
+	testInjector.register("createCommand", CreateProjectCommand);
+
+	return testInjector;
+}
+
+describe('Project Service Tests', () => {
```

Comment:
You are testing just the command, not the project Service itself, so you do not need this `describe`.
Please remove it and move the `beforeEach` and `testInjector` in the other `describe`.